### ISSUE TITLE
log base 2 or e?

### ISSUE THREAD

**Issue by getao** [pytorch/fairseq#1553]
I remembered in fairseq, the log base is 2.

However, I checked the repo and found in some files (like the 213th line in eval_lm.py), the log base seems to be e, while in some files (like 329th line in utils.py) the log base seems to be 2.

So, what on earth is the log base in fairseq?

**Comment by myleott** [comment]
In most cases we output base 2. For example in the cross entropy criterion we convert the loss to be base 2: https://github.com/pytorch/fairseq/blob/master/fairseq/criterions/cross_entropy.py#L60

In eval_lm.py we directly take the per-position log probs from the model, which are base e.

This is super confusing though and should be better documented. We should either always indicate the base whenever we print a loss, or at least have this indicated in comments.

**Comment by getao** [comment]
> In most cases we output base 2. For example in the cross entropy criterion we convert the loss to be base 2: https://github.com/pytorch/fairseq/blob/master/fairseq/criterions/cross_entropy.py#L60
> 
> In eval_lm.py we directly take the per-position log probs from the model, which are base e.
> 
> This is super confusing though and should be better documented. We should either always indicate the base whenever we print a loss, or at least have this indicated in comments.

Thanks very much for your answer. So, the scores shown when using generate.py or interactive are base e, right?

**Comment by myleott** [comment]
Correct. Same for eval_lm.py.

**Comment by myleott** [comment]
Fix here: https://github.com/pytorch/fairseq/commit/660d69fd2bdc4c3468df7eb26b3bbd293c793f94
